Improve the Teleport Cloud Getting Started guide (#20837)
gravitational/teleport: docs/pages/choose-an-edition/teleport-cloud/getting-started.mdx

  installation instructions on [Docker's
  website](https://docs.docker.com/get-docker/).

  <Notice type="tip">

  Docker is only required for a local demo environment. If you want to get
  started registering your servers in your infrastructure with Teleport, view
  our documentation on how to [register servers and configure access to
  them](../../server-access/introduction.mdx).

  </Notice>

- The `tsh` client tool. Visit our [Installation Guide](../../installation.mdx)
  for instructions. If you want to get started quickly, you can skip this for
  now and access Teleport through your browser.

## Step 1/4. Spin up your server

Spin up a server on your workstation. In Step 2, we will register this server
with your Teleport Enterprise Cloud tenant:

```code
$ docker run -it ubuntu:22.10 /bin/bash
```

Make sure you have installed `curl` in your container:

```code
$ apt-get update
$ apt-get install -y curl
```

Keep this shell open in your terminal for the next step.

## Step 2/4. Install the Teleport SSH Service

To install the Teleport SSH Service on your server, visit the Teleport Web UI at
the address of your Teleport Enterprise Cloud tenant. For example, visit:

> https://mytenant.teleport.sh

Author: I've added a suggestion to work the variable into this paragraph without a `VarList`.

If this is your first time visiting the Web UI, you will see a prompt to add
your first resource: 

> ![Add your first

Reviewer: The images may be compressed, but they are still huge in both file size and resolution for the web:

![image](https://user-images.githubusercontent.com/2349184/215551719-d3eb99f8-bffa-47f1-9d7c-6accb238c372.png)

Consider scaling them down to something closer to the maximum width of the `DocsPage_main` element to increase load time.

Author: I've added the `@2x` marker to the filenames (https://goteleport.com/docs/contributing/documentation/reference/#image-pixel-density-markers). In any case, it looks like NextJS is converting the PNG images to Webp and lazy-loading them at reasonable sizes, so load time shouldn't be an issue

resource](../../../img/cloud/getting-started/add-my-first-resource.png)

This will take you to the Teleport Discover UI:

![Select resource
type](../../../img/cloud/getting-started/select-resource-type.png)

Leave **Server** selected and click **Next**.

You will see a script that you can copy from the Teleport Web UI:

![Configure resource](../../../img/cloud/getting-started/configure-resource.png)

Copy the script and paste it into the shell you opened in the previous step.
Remove `sudo` from the start of the script so you can run it on your container.
It should look something like this:

```text

> $ bash -c "$(curl -fsSL https://TENANT_SUBDOMAIN.teleport.sh/scripts/SCRIPT_NAME/install-node.sh)" 

Author: This is why I used the `text` label—is there an alternative that users will be less likely to think is a copy-pastable command?

> $ bash -c "$(curl -fsSL https://TENANT_SUBDOMAIN.teleport.sh/scripts/SCRIPT_NAME/install-node.sh)" 

Reviewer: My bad, I assumed it was `code` because of the `$` prompt. Possible improvements:
- Get rid of the `$` prompt
- Change it to a quote block with `>`
- Make it part of the paragraph above using inline code tics.

> $ bash -c "$(curl -fsSL https://TENANT_SUBDOMAIN.teleport.sh/scripts/SCRIPT_NAME/install-node.sh)" 

Author: Addressed in 7ae87d817481cde63103bc06f6e7780f82345413

will see a message in the Web UI that your new server connected successfully:

![Connected
successfully](../../../img/cloud/getting-started/successfully-connected.png)

Choose the OS user you will assume when you authenticate to the server, in this
case, `root`:

![Choose your user](../../../img/cloud/getting-started/set-up-access.png)

From there, you can test the connection and access the server. Click **Start
Session** to connect:

![Start session](../../../img/cloud/getting-started/test-connection.png)

<Notice type="tip">

If you get an error connecting to your container, interrupt the `teleport`
process in your container's shell with **Ctrl-C**, then run `teleport start` to
restart the process.

</Notice>

You should see a terminal prompt in your browser. Your Teleport Enterprise Cloud
account proxies your SSH connection through the Teleport Proxy Service, which
connects to your container via a reverse tunnel. 

## Step 3/4. Access your server from the command line

Log in to your Teleport Enterprise Cloud tenant via the command line, filling in
the URL of your tenant and the name of your Teleport user:

```code

> $ tsh login --proxy=<Var name="tenant-url" /> --user=<Var name="username" />

Reviewer: since `teleport.sh` is determined, should we reduce the scope of the Var?

```

List the servers your Teleport user can access. You should see the name of the
container you just registered:

```code

> $ tsh ls

Reviewer: Why remove the example output? Should we include something like:

```
Node Name    Address    Labels
------------ ---------- ----------------------------------------------------------------------------------------
(=some.var=) ⟵ Tunnel   hostname=(=some.var=),teleport.internal/resource-id=(=some.other.var=)
```

So that first time users know what to expect?

```

Access your server as the `root` user:

```code

> $ tsh ssh root@<Var name="server-name" />

Reviewer: Per the previous note, if we include the output then we could change this Var name to "node-name" to make it clearer what value from the output belongs here.

```

Run some commands in your terminal. In the next step, we will show you how you
can play back your session. When you have finished, run `exit` to end your
session.

## Step 4/4. Play back your session

While Teleport Enterprise Cloud proxies SSH connections to registered servers,

> it stores recordings of SSH sessions so you can play them back later to

Reviewer: Consider:

```suggestion
it records SSH sessions so you can play them back later to
```

In the current form the "While" could be read as the second conjunction form, indicating a contrast. IMHO this indicates the non-contrasting conjunction better.

investigate issues.

> In the Teleport Web UI, click the **Activity** tab in the sidebar on the left of

Reviewer: FYI We never had the reader click the **FINISH** button during the previous process, so if they are following the guide exactly they don't yet see this button. Even after "Finish" they need to click **BROWSE EXISTING RESOURCES** to see what's described here.

the screen, then click **Ression Recordings**. You will see your session from
the previous step:

![Session

> recordings](../../../img/cloud/getting-started/session-recordings.png)

Reviewer: When following this guide exactly, the reader should have two recording listed; one for the connection through the web UI and one for the CLI-session where we had them run the commands we want to review.
This image should reflect that, and the instructions below should specify the second session (which will be first in the list with default ordering, something else to point out) with the commands we want to review.

Author: I've reorganized the guide to put the session recording section after the Web UI section, so hopefully this makes more sense.

Click **PLAY**. You will see a full recording of your session.

## Next steps

Now that you have registered your first server with Teleport, read about how you

> can register the rest of your infrastructure, including:

Reviewer: This sentence sounds funny to me; this guide has the user add a test resource to the cluster but this sentence treats it as if it's just another part of the actual infrastructure the user wants to keep in Teleport.

Consider linking to instructions (or briefly describe, if non-existent), for removing the test node from the cluster, and rephrase along the lines of "begin adding proper infrastructure for your environment to Teleport".

Reviewer: Also, following Next Steps conventions, consider making each item in the section a bullet point, with the current points being second-level.

traffic to the server from authorized clients.

## Prerequisites

- A Teleport Enterprise Cloud account. You can sign up for a free trial at the
  following page:

  https://goteleport.com/signup/

- Docker installed on your workstation. We will use a Docker container to
  register a server with Teleport via the Teleport SSH Service. You can find
  installation instructions on [Docker's
  website](https://docs.docker.com/get-docker/).

  <Notice type="tip">

  Docker is only required for a local demo environment. If you want to get
  started registering your servers in your infrastructure with Teleport, view
  our documentation on how to [register servers and configure access to
  them](../../server-access/introduction.mdx).

  </Notice>

- The `tsh` client tool. Visit our [Installation Guide](../../installation.mdx)
  for instructions. If you want to get started quickly, you can skip this for
  now and access Teleport through your browser.

## Step 1/4. Spin up your server

Spin up a server on your workstation. In Step 2, we will register this server
with your Teleport Enterprise Cloud tenant:

```code
$ docker run -it ubuntu:22.10 /bin/bash
```

> In the shell you opened within the `ubuntu:22.10` container, make sure you have

Reviewer: ```suggestion
This will bring your terminal session into the `ubuntu:22.10` container. From there, make sure you have
```

Reviewer: Doesn't have to be this specifically, I just wanted to change the language to explain that the same session you're already in is now operating from a different environment.
